pkg/reconciler/common/releases.go

```diff
@@ -46,20 +47,35 @@ func TargetVersion(instance v1alpha1.TektonComponent) string {
 
 // TargetManifest returns the manifest for the TargetVersion
 func TargetManifest(instance v1alpha1.TektonComponent) (mf.Manifest, error) {
-	return Fetch(manifestPath(TargetVersion(instance), instance))
+	return FetchRecursive(manifestPath(TargetVersion(instance), instance))
 }
 
-func Fetch(path string) (mf.Manifest, error) {
+// fetchWithCache is a generic function to fetch manifest with caching
+func fetchWithCache(path string, cache map[string]mf.Manifest, fetchFn func(string) (mf.Manifest, error)) (mf.Manifest, error) {
 	if m, ok := cache[path]; ok {
 		return m, nil
 	}
-	result, err := mf.NewManifest(path)
+	result, err := fetchFn(path)
 	if err == nil {
 		cache[path] = result
 	}
 	return result, err
 }
 
+// Fetch returns a manifest from the given path only, not recursively
+func Fetch(path string) (mf.Manifest, error) {
```

Comment:
We are using Fetch in test files only and can we use FetchRecursive in place for Fetch?

Reply:
I see that the current usage of `Fetch` is always with the full file path as the filename, and even if changed to FetchRecursive, there is no difference, so it wasn’t replaced.
Additionally, the `Fetch` method is kept solely for backward compatibility, to avoid breaking other libraries that use this public method, preventing breaking changes after future upgrades.

Reply:
LGTM